File: БР1.2/Smirnova_Karina/labs/lab1/src/main/java/org/renting/rentingservice/controller/UserController.java
package org.renting.rentingservice.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.renting.rentingservice.dto.user.UpdateMeRequest;
import org.renting.rentingservice.dto.user.UserMeResponse;
import org.renting.rentingservice.dto.user.UserProfileResponse;
import org.renting.rentingservice.security.SecurityUtils;
import org.renting.rentingservice.service.AuthService;
import org.renting.rentingservice.service.UserService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@Tag(name = "Users", description = "Профиль и настройки текущего пользователя")
public class UserController {

    private final UserService userService;
    private final AuthService authService;

    @GetMapping("/users/me")
    @Operation(summary = "Мой профиль", description = "Возвращает данные текущего пользователя")
    public UserMeResponse getMe() {
        return userService.getMe(SecurityUtils.currentUserId());
    }

    @PatchMapping("/users/me")
    @Operation(summary = "Обновить профиль", description = "Обновляет данные текущего пользователя")
    public UserMeResponse updateMe(@Valid @RequestBody UpdateMeRequest request) {
        return userService.updateMe(SecurityUtils.currentUserId(), request);
    }

    @PostMapping("/users/me/email/verify")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    @Operation(summary = "Повторная отправка email", description = "Генерирует новый токен подтверждения и отправляет письмо")
    public void resendVerifyEmail() {
        authService.resendVerification(SecurityUtils.currentUserId());
    }

    @GetMapping("/users/me/profile")
    @Operation(summary = "Публичный профиль", description = "Возвращает профиль текущего пользователя с его недвижимостью")
    public UserProfileResponse getProfile() {
        return userService.getProfile(SecurityUtils.currentUserId());
    }
}

File: БР1.2/Цатинян Артём/labs/lab2/identity-service/src/main/kotlin/ru/itmo/restaurantbooking/lab2/identity/adapter/rest/UserController.kt
package ru.itmo.restaurantbooking.lab2.identity.adapter.rest

import io.swagger.v3.oas.annotations.Parameter
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ru.itmo.restaurantbooking.lab2.identity.adapter.rest.dto.UpdateProfileRequest
import ru.itmo.restaurantbooking.lab2.identity.adapter.rest.dto.UserProfileResponse
import ru.itmo.restaurantbooking.lab2.identity.service.AuthService
import ru.itmo.restaurantbooking.lab2.identity.service.UserService

@RestController
@RequestMapping("/api/v1/users")
class UserController(
    private val userService: UserService,
    private val authService: AuthService
) {
    @GetMapping("/me")
    fun me(
        @Parameter(hidden = true)
        @RequestHeader("Authorization", required = false) authorization: String?
    ): UserProfileResponse =
        userService.profile(authService.authenticate(authorization).id)

    @PatchMapping("/me")
    fun updateMe(
        @Parameter(hidden = true)
        @RequestHeader("Authorization", required = false) authorization: String?,
        @Valid @RequestBody request: UpdateProfileRequest
    ): UserProfileResponse = userService.updateProfile(authService.authenticate(authorization).id, request)
}
